File: leetcode/weekly420/Q2.cpp
class Solution {
public:
    int numberOfSubstrings(string s, int k) {
        int n = s.length(), result = 0;
        
        for(int len = 1; len <= n; len++) {
            vector<int> freq(26, 0);
            
            for(int i = 0; i < len; i++) {
                freq[s[i] - 'a']++;
            }
            
            if(hasCharWithFreqK(freq, k)) result++;
            
            for(int i = len; i < n; i++) {
                freq[s[i] - 'a']++;
                freq[s[i-len] - 'a']--;
                if(hasCharWithFreqK(freq, k)) result++;
            }
        }
        return result;
    }
    
private:
    bool hasCharWithFreqK(vector<int>& freq, int k) {
        for(int f : freq) {
            if(f >= k) return true;
        }
        return false;
    }
};

File: leetcode/weekly426/Q2.py
class Solution:
    def getLargestOutlier(self, nums: List[int]) -> int:
        length = len(nums)
        total = sum(nums)

        counts = {}

        for num in nums:
            counts[num] = counts.get(num, 0) + 1

        result = float('-inf')

        for num in nums:
            remaining = total - num

            if remaining % 2 == 0:
                target = remaining // 2
                if target in counts:
                    if counts[target] > 1 or (target != num and counts[target] > 0):
                        result = max(result, num)

        return result